plugin_sge.py

```diff
@@ -228,71 +228,74 @@ def _get_statq_from_xml(fn):
     def _get_worker_nodes_from_xml(self, fn, write_method=options.write_method):
         logging.debug('Parsing tree of %s' % fn)
         anonymize = anonymize_func()
+
         with open(fn, 'rb') as fin:
             tree = etree.parse(fin)
-            root = tree.getroot()
-            worker_nodes = list()
-            existing_node_names = set()
-            for queue_elem in root.findall('queue_info/Queue-List'):
-                worker_node = dict()
-                resources = queue_elem.findall('resource')
-                # TODO: find a way to loop ONCE for both hostname and qname!!
-                try:
-                    slots_used = int(queue_elem.find('./slots_used').text)
-                except AttributeError:
-                    slots_used = 0
-                count = 0
-                for resource in resources:
-                    if resource.attrib.get('name') == 'hostname':
-                        worker_node['domainname'] = resource.text if not options.ANONYMIZE else anonymize(resource.text, 'wns')
-                        count += 1
-                        if count == 2: break
-                    elif resource.attrib.get('name') == 'qname':
-                        if not slots_used:
-                            worker_node['qname'] = set()
-                        else:
-                            worker_node['qname'] = set(resource.text[0]) \
-                                if not options.ANONYMIZE else set(anonymize(resource.text[0], 'qs'))
-                        count += 1
-                        if count == 2: break
-                else:
-                    raise ValueError("No such resource")
+ 
```

Comment:
bad diff display: I'm only pushing the rest of this code outside the with context manager, no need to be in there